lib/util/state.js

```diff
@@ -304,6 +304,103 @@ export function getActiveItinerary (otpState) {
     : null
 }
 
+/**
+ * Returns true if the OTP state contains a response within the current active search.
+ * @param {*} state The entire redux state used to obtain the response.
+ */
+function hasResponse (state) {
+  const activeSearch = getActiveSearch(state.otp)
+  return !!(activeSearch && activeSearch.response)
+}
+
+/**
+ * Returns true if the OTP state contains one or more OTP responses for the current active search.
+ * @param {*} state The entire redux state used to obtain the responses.
+ */
+function hasAtLeastOneResponse (state) {
+  const activeSearch = getActiveSearch(state.otp)
+  return activeSearch && activeSearch.response && activeSearch.response.length > 0
+}
+
+/**
+ * Returns the raw OTP response for the current active search.
+ * @param {*} state The entire redux state used to obtain the response.
+ */
+function getOtpResponse (state) {
+  const activeSearch = getActiveSearch(state.otp)
+  return activeSearch && activeSearch.response && activeSearch.response.otp
+}
+
+/**
+ * Returns true if the query routing type specifies 'ITINERARY'.
+ * @param {*} state The entire redux state in which the query is stored.
+ */
+function queryIsItineraryRouting (state) {
+  const activeSearch = getActiveSearch(state.otp)
+  return activeSearch &&
+    activeSearch.query &&
+    activeSearch.query.routingType === 'ITINERARY'
+}
+
+/**
+ * Returns the visible itinerary to render on the map and in narrative components.
+ * @param {*} state The entire redux state from which to retrieve the itinerary.
+ */
+export function getVisibleItinerary (state) {
+  const activeSearch = getActiveSearch(state.otp)
+  return activeSearch && activeSearch.visibleItinerary
+}
+
+/**
+ * Returns the itinerary to be passed to the transitive overlay for rendering.
+ * @param {*} state The entire redux state from which to derive the itinerary.
+ */
+function getItineraryToRender (state) {
+  const visibleItinerary = getVisibleItinerary(state)
+  const activeItinerary = getActiveItinerary(state.otp)
+  const itins = getActiveItineraries(state.otp)
+
+  const visibleIndex = visibleItinerary !== undefined && visibleItinerary !== null
+    ? visibleItinerary
+    : activeItinerary
+  return itins[visibleIndex] || activeItinerary
+}
+
+/**
+ * Converts an OTP itinerary to the transitive.js format,
+ * using a selector to prevent unnecessary re-renderings of the transitive overlay.
+ */
+export const getTransitiveData = createSelector(
+  hasResponse,
+  hasAtLeastOneResponse,
+  getOtpResponse,
+  queryIsItineraryRouting,
```

Comment:
queryIsItineraryRouting and hasAtLeastOneResponse could just be joined into a single method called itineraryResponseExists. No need to maintain two separate methods when they are used in an "and" relationship in the conditional.

Reply:
Combined in https://github.com/opentripplanner/otp-react-redux/pull/366/commits/344fc9fb80b4de8724aa55b4bfae5dafd15d0fc9.